hack/make-rules/test-cmd-completion.sh

```diff
@@ -0,0 +1,97 @@
+#!/bin/bash
+
+# Copyright 2018 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# This command checks that the built commands can function together for
+# simple scenarios.  It does not require Docker.
+
+check_completion_result(){
+    local completion COMP_CWORD COMP_LINE COMP_POINT COMP_WORDS COMPREPLY=()
+
+    # load bash-completion if necessary
+    declare -F _completion_loader &>/dev/null || {
+        source /usr/share/bash-completion/bash_completion
+    }
+
+    RESULT="$1"
+    shift
+
+    COMP_LINE=$@
+    COMP_POINT=${#COMP_LINE}
+    eval set -- "${@}"
+    COMP_WORDS=("${@}")
+
+    # add '' to COMP_WORDS if the last character of the command line is a space
+    [[ ${COMP_LINE[@]: -1} = ' ' ]] && COMP_WORDS+=('')
```

Comment:
The `:-1` here is not an index but a default since you can array index and string index at once. Similarly here, more verbosity isn't bad:

```sh
local full_line
full_line="${COMP_LINE[@]}"
if [[ "${full_line:(-1)}" = ' ' ]]; then
    COMP_WORDS+=('')
fi
```